setup.pl

```diff
@@ -143,7 +143,9 @@
 			system("$config_directory/stop.bat >/dev/null 2>&1");
 			}
 		else {
-			system("$config_directory/stop >/dev/null 2>&1");
+			if (-r "$config_directory/.pre-install") {
```

Comment:
Also, call conditional `.pre-install` to handle old and new systems (with `systemd`) correctly.